packages/@coorpacademy-components/src/molecule/base-modal/index.js

```diff
@@ -0,0 +1,152 @@
+              }}
+            />
+          ) : null}
+          {confirmButton?.onConfirm && confirmButton?.label ? (
+            <ButtonLink
+              {...{
+                customStyle: {
+                  width: '121px'
+                },
+                type: 'primary',
+                onClick: confirmButton.onConfirm,
+                label: confirmButton?.label,
+                disabled: confirmButton?.disabled,
+                ...(confirmButton?.iconName
+                  ? {
+                      icon: {
+                        position: 'left',
+                        faIcon: {
+                          name: confirmButton?.iconName,
+                          color: '#FFFFFF',
+                          size: 16
+                        }
+                      }
+                    }
+                  : null)
+              }}
+            />
+          ) : null}
+        </div>
+        {footerDescription?.text ? (
+          <div className={footerDescription.isError && style.footerDescriptionError}>
+            {footerDescription.text}
+          </div>
+        ) : null}
+      </div>
+    );
+  }, [footer, cancelButton, confirmButton, footerDescription]);
+
+  if (!isOpen || !title || !children) return null;
+
+  function handleOnClose(e) {
+    e?.stopPropagation();
+    onClose();
+  }
+
+  return (
+    <div className={style.modalWrapper}>
+      <div className={style.modal}>
+        <header className={style.header}>
+          {headerIcon && headerIcon.name ? (
+            <div className={style.headerIcon}>
+              <Icon
+                iconName={headerIcon.name}
+                iconColor={headerIcon.color}
+                backgroundColor={headerIcon.backgroundColor}
+                size={{
+                  faSize: 20,
+                  wrapperSize: 48
+                }}
+              />
+            </div>
+          ) : null}
+          <div className={style.headerContent}>
+            <div className={style.headerTitle}>{title}</div>
+            {description ? <div className={style.headerDescription}>{description}</div> : null}
+          </div>
+          <div className={style.headerCloseIcon} onClick={handleOnClose}>
+            <Icon
+              iconName="close"
+              backgroundColor="#F4F4F5"
+              size={{
+                faSize: 14,
+                wrapperSize: 28
+              }}
+            />
+          </div>
+        </header>
+        <div className={style.body}>{children}</div>
+        <Footer />
+      </div>
+    </div>
+  );
+};
+
+BaseModal.propTypes = {
+  title: PropTypes.string,
+  headerIcon: PropTypes.shape({
+    name: PropTypes.string,
+    color: PropTypes.string,
+    backgroundColor: PropTypes.string
+  }),
+  description: PropTypes.string,
+  children: PropTypes.node,
+  isOpen: PropTypes.bool,
+  footer: PropTypes.func,
+  footerDescription: PropTypes.shape({
+    text: PropTypes.string,
+    isError: PropTypes.boolean
+  }),
```

Comment:
what about footer either function, or the text/isError shape?
(as both are not supposed to be called together)

Reply:
```
footerDescription: PropTypes .oneOfType({
   PropTypes.func
   PropTypes.shape({
    text: PropTypes.string,
    isError: PropTypes.boolean,
    cancelButton:  ....,
   confirmButton: ....
  })
})
````